include/range/v3/action/slice.hpp

```diff
@@ -86,11 +139,11 @@ namespace ranges
                     CONCEPT_ASSERT_MSG(ErasableRange<Rng, I, I>(),
                         "The object on which action::slice operates must allow element "
                         "removal.");
-                    CONCEPT_ASSERT_MSG(meta::and_<ConvertibleTo<T, range_difference_type_t<Rng>>,
-                            ConvertibleTo<U, range_difference_type_t<Rng>>>(),
-                        "The bounds passed to action::slice must be convertible to the range's "
-                        "difference type. TODO slicing from the end with 'end-2' syntax is not "
-                        "supported yet, sorry!");
+                    CONCEPT_ASSERT_MSG(meta::or_<BidirectionalRange<Rng>,
+                        meta::negate<meta::or_<is_end_type<T>, is_end_type<U>>>>(),
+                        "To slice from the end the object on which action::slice operates it must be a model"
+                        "of the BidirectionalRange concept.");
```

Comment:
I'd just say "The range upon which action::slice operates must model BidrectionalRange to slice from the end." here. (Modulo how you choose to address the earlier comment about this requirement being insufficient, that is ;))